comp/collector/collector/collectorimpl/collector_demux_test.go

```diff
@@ -36,16 +37,55 @@ import (
 type CollectorDemuxTestSuite struct {
 	suite.Suite
 
-	demux demultiplexer.FakeSamplerMock
-	c     *collectorImpl
+	demux             demultiplexer.FakeSamplerMock
+	c                 *collectorImpl
+	SenderManagerMock *SenderManagerProxy
+}
+
+// SenderManagerProxy is a proxy that includes a channel to enforce
+// synchronization in the DestroySender method and implements all the methods of
+// a normal sender
+type SenderManagerProxy struct {
+	destroyChannel     chan checkid.ID
+	innerSenderManager sender.SenderManager
+}
+
+// NewSenderManagerMock creates a new instance of SenderManagerProxy
+func NewSenderManagerMock(s sender.SenderManager) *SenderManagerProxy {
+	return &SenderManagerProxy{
+		destroyChannel:     make(chan checkid.ID),
+		innerSenderManager: s,
+	}
+}
+
+// GetSender returns a sender.Sender with passed ID
+func (s *SenderManagerProxy) GetSender(id checkid.ID) (sender.Sender, error) { //nolint:revive // TODO fix revive unused-parameter
```

Comment:
You should be able to remove all the revive TODOs

Reply:
Done!